File: Java/CWE-352.java
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;

@WebServlet("/transfer")
public class TransferServlet extends HttpServlet {
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        // Получаем параметры запроса
        String fromAccount = request.getParameter("fromAccount");
        String toAccount = request.getParameter("toAccount");
        String amount = request.getParameter("amount");

        // Выполняем перевод средств (условно, без реальной логики)
        // В реальном приложении здесь будет логика работы с базой данных или API
        performTransfer(fromAccount, toAccount, amount);

        // Отправляем ответ клиенту
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        out.println("<html><body>");
        out.println("<h2>Transfer successful!</h2>");
        out.println("<p>Transferred " + amount + " from " + fromAccount + " to " + toAccount + ".</p>");
        out.println("</body></html>");
    }

    private void performTransfer(String fromAccount, String toAccount, String amount) {
        // Здесь должна быть логика перевода средств между счетами
        System.out.println("Transferring " + amount + " from " + fromAccount + " to " + toAccount);
    }

    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        out.println("<html><body>");
        out.println("<form action='transfer' method='post'>");
        out.println("From Account: <input type='text' name='fromAccount'><br>");
        out.println("To Account: <input type='text' name='toAccount'><br>");
        out.println("Amount: <input type='text' name='amount'><br>");
        out.println("<input type='submit' value='Transfer'>");
        out.println("</form>");
        out.println("</body></html>");
    }
}

File: python/CWE-352.py
from flask import Flask, request, redirect, url_for

app = Flask(__name__)

# Пример пользователя
users = {"admin": "password"}

@app.route('/')
def home():
    return '''
        <h1>Welcome</h1>
        <form action="/login" method="post">
            <p><input type=text name=username>
            <p><input type=password name=password>
            <p><input type=submit value=Login>
        </form>
        <form action="/transfer" method="post">
            <p><input type=text name=to_user placeholder="Recipient">
            <p><input type=text name=amount placeholder="Amount">
            <p><input type=submit value=Transfer>
        </form>
    '''

@app.route('/login', methods=['POST'])
def login():
    username = request.form['username']
    password = request.form['password']
    if username in users and users[username] == password:
        return redirect(url_for('home'))
    return 'Invalid credentials'

@app.route('/transfer', methods=['POST'])
def transfer():
    to_user = request.form['to_user']
    amount = request.form['amount']
    # Здесь должна быть логика перевода средств
    return f'Transferred {amount} to {to_user}'

if __name__ == '__main__':
    app.run(debug=True)
